lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1462,6 +1461,80 @@ def test_message_for_audit_mode(self):
             u'You are enrolled in the audit track for this course. The audit track does not include a certificate.'
         )
 
+    def test_invalidated_cert_data(self):
+        """
+        Verify that invalidated cert data is returned if cert is invalidated.
+        """
+        generated_certificate = self.generate_certificate(
+            "http://www.example.com/certificate.pdf", "honor"
+        )
+
+        CertificateInvalidationFactory.create(
+            generated_certificate=generated_certificate,
+            invalidated_by=self.user
+        )
+        # Invalidate user certificate
+        generated_certificate.invalidate()
+        response = views._get_cert_data(self.user, self.course, self.course.id, True, CourseMode.HONOR)
+        self.assertEqual(response['cert_status'], 'invalidated')
+        self.assertEqual(response['title'], 'Your certificate has been invalidated')
+
+    def test_downloadable_get_cert_data(self):
+        """
+        Verify that downloadable cert data is returned if cert is downloadable.
+        """
+        self.generate_certificate(
+            "http://www.example.com/certificate.pdf", "honor"
+        )
+        with patch('certificates.api.certificate_downloadable_status', Mock(
+                return_value=self.mock_certificate_downloadable_status(is_downloadable=True))):
```

Comment:
Nit: I think you can leave out the `Mock` here, and just use `patch('certificates...', return_value=...)`

From the docs: "patch takes arbitrary keyword arguments. These will be passed to the Mock (or new_callable) on construction."